src/main/java/net/neoforged/neoforge/transfer/handlers/templates/ResourceStorage.java

```diff
@@ -0,0 +1,125 @@
+package net.neoforged.neoforge.transfer.handlers.templates;
+
+import net.minecraft.core.component.DataComponentType;
+import net.neoforged.neoforge.attachment.AttachmentHolder;
+import net.neoforged.neoforge.attachment.AttachmentType;
+import net.neoforged.neoforge.transfer.IResource;
+import net.neoforged.neoforge.transfer.ResourceStack;
+import net.neoforged.neoforge.transfer.TransferAction;
+import net.neoforged.neoforge.transfer.context.IItemContext;
+import net.neoforged.neoforge.transfer.handlers.IResourceHandler;
+import net.neoforged.neoforge.transfer.handlers.ResourceStorageContents;
+
+public abstract class ResourceStorage<T extends IResource> implements IResourceHandler<T> {
```

Comment:
I am not sure that this abstraction is worth it. I'll have to look further into it.